onnxruntime/core/optimizer/transpose_optimizer/transpose_optimizer.cc

```diff
@@ -968,6 +968,13 @@ static void PermuteInput(api::GraphRef& graph, api::NodeRef& node, size_t i, con
 }
 
 static bool HandleResize(HandlerArgs& args) {
+  // TODO: Only the CPU Resize kernel handles NHWC input.
+  // Adjust this restriction once the other EPs' Resize
+  // kernel(s) supports NHWC input.
+  if (args.node.GetExecutionProviderType() != "CPUExecutionProvider") {
+    return false;
```

Comment:
Needs some more thought as the TransposeOptimizer is a L1 optimizer and there is no partitioning info at this point. But, we do need to handle the CUDA EP case as the CUDA Resize kernel is implemented assuming the provided input is NCHW.
CC:@ [yihonglyu](https://github.com/yihonglyu). Any comment ? This causes a regression for CUDA EP users in 1.12.